### PR TITLE
GDriveFileSystem: Raise `FIleNotFoundError` on `ls`.

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc-gdrive/issues/29

### CODE REVIEW

File: pydrive2/fs/spec.py [hunk 1]
```diff
@@ -434,7 +434,7 @@ def ls(self, path, detail=False):
             dir_ids = self._path_to_item_ids(base)
 
         if not dir_ids:
-            return None
+            raise FileNotFoundError
```
Comment (efiop) [inline pydrive2/fs/spec.py:437]:
```suggestion
            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), path)
```

File: pydrive2/test/test_fs.py [hunk 2]
```diff
@@ -138,6 +138,11 @@ def by_name(details):
     assert dirs == expected
 
 
+def test_ls_non_existing_dir(fs, remote_dir):
+    with pytest.raises(FileNotFoundError):
+        fs.ls(remote_dir + "dir/")
```
[the remark this answers is not in the document]
Comment (skshetry) [reply pydrive2/test/test_fs.py:143]: An assertion on `test_ls` should be enough.

[the remark this answers is not in the document]
Comment (skshetry) [reply pydrive2/test/test_fs.py:143]: See https://github.com/iterative/PyDrive2/pull/284.

File: pydrive2/test/test_fs.py [hunk 3]
```diff
@@ -116,8 +116,11 @@ def test_rm(fs, remote_dir):
     assert not fs.exists(remote_dir + "/dir/c/a")
 
 
-def test_ls(fs, remote_dir):
-    fs.mkdir(remote_dir + "dir/")
+def test_ls(fs: GDriveFileSystem, remote_dir):
+    _, base = fs.split_path(remote_dir + "dir/")
+    fs._path_to_item_ids(base, create=True)
+    assert fs.ls(remote_dir + "dir/") == []
```
Comment (skshetry) [inline pydrive2/test/test_fs.py:122]: Turns out, `fs.mkdir()` is not implemented in GDriveFileSystem. I was not sure if it was needed as it has been working fine for so long. So I only fixed in test to create a directory.

**Comment by skshetry** [comment]
Investigating some failures in https://github.com/iterative/dvc-gdrive/pull/28. Will merge this after they turn green.

**Comment by skshetry** [comment]
Looks like `find()` is also not working correctly. Will look into it next week.

**Comment by daavoo** [comment]
> Looks like `find()` is also not working correctly. Will look into it next week.

What is the failure or case where is not working?

**Comment by skshetry** [comment]
> > Looks like `find()` is also not working correctly. Will look into it next week.
> 
> What is the failure or case where is not working?

See https://github.com/iterative/dvc-gdrive/pull/28. The following returns an empty list even though the file exists, failing `status` checks.

```python
>>> fs.find("root/<>/files/md5")
[]
```

I am still trying to understand internals of GDriveFileSystem on how to fix it. 

This piece of code is problematic:

https://github.com/iterative/PyDrive2/blob/a5dc1d9a4da73f8b4172c1020726bd457fb62213/pydrive2/fs/spec.py#L476-L482

**Pull request merged**